ios/FluentUI/SegmentedControl/SegmentedControl.swift

```diff
@@ -278,32 +333,19 @@ open class SegmentedControl: UIView, TokenizedControlInternal {
     open override func layoutSubviews() {
         super.layoutSubviews()
 
-        guard !isAnimating else {
-            return
-        }
+        stackView.layoutIfNeeded()
 
-        var rightOffset: CGFloat = 0
-        var leftOffset: CGFloat = 0
-        for (index, button) in buttons.enumerated() {
-            if shouldSetEqualWidthForSegments {
-                rightOffset = ceil(CGFloat(index + 1) / CGFloat(buttons.count) * pillContainerView.frame.width)
-            } else {
-                let maxSize = CGSize(width: CGFloat.greatestFiniteMagnitude,
-                                     height: CGFloat.greatestFiniteMagnitude)
-                rightOffset = leftOffset + ceil(button.sizeThatFits(maxSize).width)
-            }
-            button.frame = CGRect(x: leftOffset,
-                                  y: 0,
-                                  width: rightOffset - leftOffset,
-                                  height: pillContainerView.frame.height)
-            leftOffset = rightOffset
-        }
-
-            // flipSubviewsForRTL only works on direct children subviews
-            pillContainerView.flipSubviewsForRTL()
+        // flipSubviewsForRTL only works on direct children subviews
+        pillContainerView.flipSubviewsForRTL()
 
         flipSubviewsForRTL()
         layoutSelectionView()
+
+        updateGradientMaskColors()
+        CATransaction.begin()
+        CATransaction.setDisableActions(true)
+        gradientMaskLayer.frame = layer.bounds
```

Comment:
Why do we need to wrap these in a CATransaction?

Reply:
Ah, looks like we don't. I think I needed them when I was just drawing the gradients, but that doesn't seem to be the case anymore. Thanks!